tests/test_parser.py

```diff
@@ -78,3 +80,48 @@ def test_symref(ref: str, expected: Attribute | None):
     parsed_ref = parser.try_parse_ref_attr()
 
     assert parsed_ref == expected
+
+
+@irdl_op_definition
+class MultiRegionOp(Operation):
+    name = "test.multi_region"
+    r1: Region
+    r2: Region
+
+
+def test_parse_multi_region_mlir():
+    ctx = MLContext()
+    ctx.register_op(MultiRegionOp)
+
+    op_str = """
+    "test.multi_region" () ({
+        ^bb0:
+    }, {
+        ^bb1:
+    }) : () -> ()
+    """
+
+    parser = MLIRParser(ctx, op_str)
+
+    op = parser.parse_op()
+
+    assert len(op.regions) == 2
+
+
+def test_parse_multi_region_xdsl():
+    ctx = MLContext()
+    ctx.register_op(MultiRegionOp)
+
+    op_str = """
+    "test.multi_region" () {
+        ^bb0:
```

Comment:
General though about this: This does not roundtrip, right? If you print this again, the `bb0` will not be printed, right?

IMO, I would prefer to have a test that roundtrips.

Reply:
I wanted to have a unit test here that tests specifically this part of the parser. I could also add interop tests with MLIR, but I'm not sure how I could test that without also testing a bunch of unrelated stuff.

I don't see how a round-trip test would benefit us here.

Reply:
The reason why we did not catch this is because we don't have an interoperation test with something that takes multiple regions, right? So that would be the test we initially should have had but forgot.

Reply:
As I said, this is more of a general thought. IMO, having IR that does not roundtrip is not great in general. Using such IR in tests feels weird to me. I would prefer to hide this behavior instead of deliberately using it here.

Reply:
Is this better?